### ISSUE TITLE
Swagger stackoverflow with recursive specs 

### ISSUE THREAD

**Issue by ikitommi** [metosin/spec-tools#75]
copied from: https://github.com/metosin/compojure-api/issues/339

---8<---
When I set the coercion to `:spec` and use a recursive spec like :
```
(s/def ::user nil?) ;; will be redefined below
(s/def ::name string)
(s/def ::parent (s/nilable ::user))
(s/def ::user (s/keys :req-un [::name ::parent]))
```
Swagger fail with this error:
```
{"type":"unknown-exception","class":"java.lang.StackOverflowError"}
```

It seems that the clojure.spec -> swagger conversion keep walking the recursive spec until the stack blows up.

**Comment by ikitommi** [comment]
Currently, we haven't use schema references in JSON Schema or in Swagger JSON Schema. This is the reason to implement those. When it works, there could be a option how to generate the nested schemas:

a) inline 'em all (current)
b) only references
c) custom behaviour?

**Comment by ikitommi** [comment]
There is a quick fix emerging in #82. Real fix would be to introduce a new `SpecReference` type for this?
